packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/util/iterateLinkedObjects.ts

```diff
@@ -43,8 +43,11 @@ export async function* iterateLinkedObjects<T extends OntologyObject>(
       },
     );
 
+    const { targetType } =
+      client.ontology.objects[sourceApiName].links[linkTypeApiName];
+
     for (const object of page.data) {
-      yield convertWireToOsdkObject(client, linkTypeApiName, object);
+      yield convertWireToOsdkObject(client, targetType, object) as unknown as T;
```

Comment:
I'm a little sad about the `as unknown` cast here

Reply:
Would writing it as: 
`convertWireToOsdkObject<T["__apiName">, typeof client.OntologyDefinition>` or something along those lines be better?

Reply:
That didn't work either, the generics for T are unassignable, probably because this is a seam between `<T extends OntologyObject>` and the OntologyDefinition side.

Reply:
Ah I see, I think this is fine for now then and we can come back to it?